cocos/tween/tween-action.ts

```diff
@@ -217,18 +275,48 @@ export class TweenAction extends ActionInterval {
             const end = prop.end;
+
             if (typeof start === 'number') {
                 prop.current = interpolation(start, end, prop.current, time);
             } else if (typeof start === 'object') {
                 // const value = prop.value;
-                for (const k in start) {
-                    // if (value[k].easing) {
-                    //     time = value[k].easing(t);
-                    // }
-                    // if (value[k].progress) {
-                    //     interpolation = value[k].easing(t);
-                    // }
-                    prop.current[k] = interpolation(start[k], end[k], prop.current[k], time);
+                if (prop.type && prop.type === 'color') {
+                    prop.current.r = interpolation(start.r, end.r, prop.current.r, time);
+                    prop.current.g = interpolation(start.g, end.g, prop.current.g, time);
+                    prop.current.b = interpolation(start.b, end.b, prop.current.b, time);
+                    prop.current.a = interpolation(start.a, end.a, prop.current.a, time);
+                } else if (prop.type && prop.type === 'rect') {
+                    prop.current.xMin = interpolation(start.xMin, end.xMin, prop.current.xMin, time);
+                    prop.current.yMin = interpolation(start.yMin, end.yMin, prop.current.yMin, time);
+                    prop.current.z = interpolation(start.z, end.z, prop.current.z, time);
+                    prop.current.w = interpolation(start.w, end.w, prop.current.w, time);
+                } else if (prop.type && prop.type === 'vec2') {
+                    prop.current.x = interpolation(start.x, end.x, prop.current.x, time);
+                    prop.current.y = interpolation(start.y, end.y, prop.current.y, time);
+                } else if (prop.type && prop.type === 'vec3') {
+                    prop.current.x = interpolation(start.x, end.x, prop.current.x, time);
+                    prop.current.y = interpolation(start.y, end.y, prop.current.y, time);
+                    prop.current.z = interpolation(start.z, end.z, prop.current.z, time);
+                } else if (prop.type && prop.type === 'vec4') {
+                    prop.current.x = interpolation(start.x, end.x, prop.current.x, time);
+                    prop.current.y = interpolation(start.y, end.y, prop.current.y, time);
+                    prop.current.z = interpolation(start.z, end.z, prop.current.z, time);
+                    prop.current.w = interpolation(start.w, end.w, prop.current.w, time);
+                } else if (prop.type && prop.type === 'size') {
+                    prop.current.width = interpolation(start.width, end.width, prop.current.width, time);
+                    prop.current.height = interpolation(start.height, end.height, prop.current.height, time);
+                } else if (prop.type && prop.type === 'quat') {
+                    Quat.slerp(prop.current, start, end, time as number);
```

Comment:
Should output some warning if there is a custom interpolation method to tell that quat does not support custom interpolation method.